### PR TITLE
feat(footer): introduction

### PR DESCRIPTION
**Pull request by asudoh**
### Related Ticket(s)

Refs #2862.

### Description

This change adds a Web Components version of footer.

### Changelog

**New**

- `<dds-footer-composite>`, for CSR the remainder of components from data.
- `<dds-footer>`, the top-level element in footer.
- `<dds-footer-logo>`, the logo in footer.
- `<dds-footer-nav>`, the primary nav in footer.
- `<dds-footer-nav-group>`, a group of links of `<dds-footer-nav>`.
- `<dds-footer-nav-item>`, a link in `<dds-footer-nav>`.
- `<dds-legal-nav>`, a secondary nav in footer.
- `<dds-legal-nav-item>`, a link in `<dds-legal-nav>`.
- `<dds-locale-button>`, a launcher for `<dds-locale-modal>`.

**Changed**

- Footer styles so it can be reused for Web Components.

### CODE REVIEW

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+##### HTML
+
+```html
+<app-footer-container></app-footer-container>
+```
+
+## `<dds-footer-composite>`
+
+`<dds-footer-composite>` automatically renders contents of footer from the given data.
+You can populate the data automatically from `@carbon/ibmdotcom-servies` by mixing a Redux store into this custom element class, shown in the example in Getting started section above.
```
Comment (jeffchew) [inline packages/web-components/src/components/footer/__stories__/README.stories.mdx:40]: We probably should simplify this (and in general all of the components we are building out), and maybe have two different versions of what most adopters will want to do (ELI5, I just need one import and drop a tag on the page and call it a day), or the adopters that do care more about what's happening under the hood.

So one version I imagine would be what most adopters would be, and just want something like:

```
## Getting Started

### JS

import '@carbon/ibmdotcom-web-components/es/components/footer/footer'; (or whatever the import is)

### HTML

<dds-footer></dds-footer>
```

Then there would be the one you might have above if they want to get a little fancier with connecting data, etc:

```
### Under the hood

... explain what imports there are and how everything is getting connected ...
```

We're going to encounter that most adopters barely can pull together an HTML page, let alone understanding Typescript, etc and trying to connect one of the primary components for IBM.com on the page like this would be a huge challenge for them.

Comment (asudoh) [reply packages/web-components/src/components/footer/__stories__/README.stories.mdx]: Thank you for your feedback @jeffchew - Updated as discussed offline.

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+// Connects `<dds-footer-composite>` to `@carbon/ibmdotcom-servies` and creates `<app-footer-container>`
+customElements.define('app-footer-container', class extends ConnectMixin(store, mapStateToProps, mapDispatchToProps)(DDSFooterComposite) {})
+```
+
+You can replace the Redux store or its property/action mappers of your own, as long as the comnination of those yields to the same data structure as the default one.
+
+##### HTML
+
+```html
+<app-footer-container></app-footer-container>
```
Comment (jeffchew) [inline packages/web-components/src/components/footer/__stories__/README.stories.mdx:56]: let's use the prefix `myapp-` to show in instanced where the container is rolled on their own.

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+##### JS
+
+```javascript
+import '@carbon/ibmdotcom-web-components/es/components/footer/footer-container';
+```
+
+##### HTML
+
+```html
+<dds-footer-container></dds-footer-container>
```
Comment (jeffchew) [inline packages/web-components/src/components/footer/__stories__/README.stories.mdx:29]: Is there an option to choose the short footer?

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+| `<dds-footer>`           | `data-autoid="dds--footer"`            | Component   |
+| `<dds-footer-nav>`       | `data-autoid="dds--footer-nav"`        | Component   |
+| `<dds-footer-nav-group>` | `data-autoid="dds--footer-nav-group"`  | Component   |
+| `<dds-footer-logo>`      | `data-autoid="dds--footer-logo"`       | Component   |
+| `<dds-locale-button>`    | `data-autoid="dds--footer-locale-btn"` | Interactive |
+| `<dds-legal-nav>`        | `data-autoid="dds--legal-nav"`         | Component   |
+
+<Description markdown={corsproxy} />
+
+## Contributing
```
Comment (jeffchew) [inline packages/web-components/src/components/footer/__stories__/README.stories.mdx:176]: Don't need the extra `## Contributing` header, since the markdown include already comes with a heading.

Comment (asudoh) [reply packages/web-components/src/components/footer/__stories__/README.stories.mdx]: I see thanks - Updated.

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx (hunk 3 above)
Comment (asudoh) [reply packages/web-components/src/components/footer/__stories__/README.stories.mdx:29]: Yes, `<dds-footer-container>` supports all attributes/properties that `<dds-footer-composite>` supports, given `<dds-footer-container>` is `<dds-footer-composite>` with the default Redux store mixed in. Made an update to reflect that.

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx (hunk 2 above)
Comment (asudoh) [reply packages/web-components/src/components/footer/__stories__/README.stories.mdx]: Makes sense, done.

File: packages/web-components/src/components/footer/__stories__/README.stories.mdx (hunk 3 above)
Comment (jeffchew) [reply packages/web-components/src/components/footer/__stories__/README.stories.mdx:29]: Thanks, can there be a separate storybook story that shows the short footer? Also might help to show in the quick start section how to configure that because I think a lot of adopters will be reaching for that.

Comment (asudoh) [reply packages/web-components/src/components/footer/__stories__/README.stories.mdx:29]: Makes sense @jeffchew - Updated.

**Comment by jeffchew** [comment]
@asudoh looks good! though the default now seems to not be loading the footer links.

**Comment by asudoh** [comment]
Great catch @jeffchew  - Fixed.

**Comment by jeffchew** [comment]
LGTM, thank you @asudoh !
